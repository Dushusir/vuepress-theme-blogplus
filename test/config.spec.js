/**
 * @jest-environment node
 */

const vuepressThemeBlog = require('../')

test('should be a function', () => {
  expect(typeof vuepressThemeBlog).toBe('function')
})

test('should match snapshot without any configuration', () => {
  const themeConfig = {}
  const config = vuepressThemeBlog(themeConfig)
  expect(themeConfig).toMatchSnapshot()
  expect(config).toMatchSnapshot()
})
