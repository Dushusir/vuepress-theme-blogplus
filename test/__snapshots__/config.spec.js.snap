// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot without any configuration 1`] = `
Object {
  "nav": Array [
    Object {
      "link": "/",
      "text": "Blog",
    },
    Object {
      "link": "/tag/",
      "text": "Tags",
    },
  ],
  "pwa": false,
  "summary": true,
  "summaryLength": 200,
}
`;

exports[`should match snapshot without any configuration 2`] = `
Object {
  "alias": Object {
    "fonts": E:\\personalProject\\vuepress-theme-blogplus\\fonts",
  },
  "define": Object {
    "THEME_BLOG_PAGINATION_COMPONENT": "Pagination",
  },
  "extendPageData": [Function],
  "plugins": Array [
    "@vuepress/plugin-nprogress",
    Array [
      "@vuepress/medium-zoom",
      true,
    ],
    Array [
      "@vuepress/search",
      Object {
        "searchMaxSuggestions": 10,
      },
    ],
    Array [
      "@vuepress/blog",
      Object {
        "directories": Array [
          Object {
            "dirname": "_posts",
            "id": "post",
            "path": "/",
          },
        ],
        "feed": undefined,
        "frontmatters": Array [
          Object {
            "id": "tag",
            "keys": Array [
              "tags",
            ],
            "path": "/tag/",
          },
        ],
        "globalPagination": Object {
          "lengthPerPage": 5,
        },
      },
    ],
    Array [
      "smooth-scroll",
      false,
    ],
  ],
}
`;
